src/SpendPermissionManager.sol

```diff
@@ -351,6 +358,9 @@ contract SpendPermissionManager is EIP712 {
     ///
     /// @param spendPermission Details of the spend permission.
     function _approve(SpendPermission memory spendPermission) internal {
+        // revert if spend permission has already been approved (regardless of revoke status)
+        if (isApproved(spendPermission)) revert ExistingSpendPermission(getHash(spendPermission));
```

Comment:
The goal of guiding apps to have unique permissions is valid, but this solution causes a side effect of race conditions. Imagining the backend logic an app will use to determine if they need to prepend an `approveWithSignature` call in their batch, it is easiest to condition on the current approval state. If it is not yet approved, then prepend the call in addition to my spend. This gets tricky though in cases where I need to submit two transactions and before the first one finalizes from the perspective of my node returning the approval state, I submit my second one with a duplicated `approveWithSignature`. This second transaction will fail because of this check and revert the other operations I am trying to do. I would generally recommend not blocking duplicate approvals explicitly because it makes transaction submission more reliable, especially in cases where other entities apply signatures.

Reply:
ah, good point, I totally agree with this. allowing dupes means the app needs to understand something about how permissions work, but doesn't necessarily lead to additional complexity in their code. failing dupes means they will eventually need the same understand AND there will be additional complexity in the code